platform/spire/BUILD.bazel

```diff
@@ -1,43 +1,30 @@
+load("@rules_python//python:defs.bzl", "py_binary")
 load("//bazel:package.bzl", "homeworld_deb", "pythonize")
-load("//bazel:zip_file.bzl", "zip_file")
 
 # TODO: have a stub test (checking whether spire -h >/dev/null exits successfully) during build
 
-clustered = {
-    "dns-addon.yaml": "//dnsmasq:kubernetes.yaml",
-    "dns-monitor.yaml": "//dns-monitor:kubernetes.yaml",
-    "flannel.yaml": "//flannel:kubernetes.yaml",
-    "flannel-monitor.yaml": "//flannel-monitor:kubernetes.yaml",
-    "user-grant.yaml": "//user-grant:kubernetes.yaml",
-}
-
-mappings = {
-    "__main__/spire/src": "",
-    "__main__/spire/resources": "resources",
-    "__main__/GIT_VERSION": "resources/GIT_VERSION",
-    "__main__/upload/keyring.gpg": "resources/homeworld-archive-keyring.gpg",
-    "__main__/upload/BRANCH_NAME": "resources/APT_BRANCH",
-    "__main__/upload/DOWNLOAD_URL": "resources/APT_URL",
-}
-
-mappings.update({
-    "__main__/" + ref.lstrip("/").replace(":", "/"): "resources/clustered/" + name
-    for name, ref in clustered.items()
-})
+py_binary(
+    name = "spirepy",
+    srcs = glob([
+        "src/*.py",
+    ]),
+    data = [
```

Comment:
what exactly does `py_binary` do with `data`?  It's not well documented

Reply:
So this is what py_binary builds into the zip file:

```
$ unzip -l bazel-bin/spire/spirepy.zip
Archive:  bazel-bin/spire/spirepy.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
    13135  2010-01-01 00:00   __main__.py
        0  2010-01-01 00:00   __init__.py
        0  2010-01-01 00:00   runfiles/__main__/__init__.py
        0  2010-01-01 00:00   runfiles/__main__/spire/src/__init__.py
        0  2010-01-01 00:00   runfiles/__main__/spire/__init__.py
      950  2010-01-01 00:00   runfiles/__main__/spire/src/__main__.py
    13162  2010-01-01 00:00   runfiles/__main__/spire/src/access.py
     6802  2010-01-01 00:00   runfiles/__main__/spire/src/authority.py
    11888  2010-01-01 00:00   runfiles/__main__/spire/src/command.py
    10181  2010-01-01 00:00   runfiles/__main__/spire/src/configuration.py
     3471  2010-01-01 00:00   runfiles/__main__/spire/src/deploy.py
     3040  2010-01-01 00:00   runfiles/__main__/spire/src/infra.py
     6479  2010-01-01 00:00   runfiles/__main__/spire/src/iso.py
     1792  2010-01-01 00:00   runfiles/__main__/spire/src/keycrypt.py
     5903  2010-01-01 00:00   runfiles/__main__/spire/src/keys.py
      653  2010-01-01 00:00   runfiles/__main__/spire/src/metadata.py
     5622  2010-01-01 00:00   runfiles/__main__/spire/src/packages.py
     1426  2010-01-01 00:00   runfiles/__main__/spire/src/query.py
      588  2010-01-01 00:00   runfiles/__main__/spire/src/resource.py
     4896  2010-01-01 00:00   runfiles/__main__/spire/src/seq.py
    11587  2010-01-01 00:00   runfiles/__main__/spire/src/setup.py
     1779  2010-01-01 00:00   runfiles/__main__/spire/src/ssh.py
     2140  2010-01-01 00:00   runfiles/__main__/spire/src/template.py
      998  2010-01-01 00:00   runfiles/__main__/spire/src/util.py
    17096  2010-01-01 00:00   runfiles/__main__/spire/src/verify.py
    22667  2010-01-01 00:00   runfiles/__main__/spire/src/virt.py
      433  2010-01-01 00:00   runfiles/__main__/dns-monitor/__init__.py
     1240  2010-01-01 00:00   runfiles/__main__/dns-monitor/kubernetes.yaml
      433  2010-01-01 00:00   runfiles/__main__/dnsmasq/__init__.py
     2853  2010-01-01 00:00   runfiles/__main__/dnsmasq/kubernetes.yaml
      433  2010-01-01 00:00   runfiles/__main__/flannel/__init__.py
     2209  2010-01-01 00:00   runfiles/__main__/flannel/kubernetes.yaml
      433  2010-01-01 00:00   runfiles/__main__/flannel-monitor/__init__.py
     4608  2010-01-01 00:00   runfiles/__main__/flannel-monitor/kubernetes.yaml
      433  2010-01-01 00:00   runfiles/__main__/spire/resources/__init__.py
      202  2010-01-01 00:00   runfiles/__main__/spire/resources/isolinux.cfg.serial
      417  2010-01-01 00:00   runfiles/__main__/spire/resources/kubeconfig-local.yaml
     1737  2010-01-01 00:00   runfiles/__main__/spire/resources/netcfg.postinst
     3716  2010-01-01 00:00   runfiles/__main__/spire/resources/postinstall.sh
      826  2010-01-01 00:00   runfiles/__main__/spire/resources/prepartition.sh
    18755  2010-01-01 00:00   runfiles/__main__/spire/resources/preseed.cfg.in
     3949  2010-01-01 00:00   runfiles/__main__/spire/resources/prometheus.yaml
     1097  2010-01-01 00:00   runfiles/__main__/spire/resources/servicemap.yaml
      831  2010-01-01 00:00   runfiles/__main__/spire/resources/setup.yaml
     1611  2010-01-01 00:00   runfiles/__main__/spire/resources/setup-schema.yaml
     1865  2010-01-01 00:00   runfiles/__main__/spire/resources/sshd_config
      433  2010-01-01 00:00   runfiles/__main__/upload/__init__.py
        7  2010-01-01 00:00   runfiles/__main__/upload/BRANCH_NAME
       47  2010-01-01 00:00   runfiles/__main__/upload/DOWNLOAD_URL
     1280  2010-01-01 00:00   runfiles/__main__/upload/keyring.gpg
      433  2010-01-01 00:00   runfiles/__main__/user-grant/__init__.py
     2726  2010-01-01 00:00   runfiles/__main__/user-grant/kubernetes.yaml
      433  2010-01-01 00:00   runfiles/__main__/version/__init__.py
       47  2010-01-01 00:00   runfiles/__main__/version/GIT_VERSION
     3561  2010-01-01 00:00   runfiles/bazel_tools/tools/python/py3wrapper.sh
---------                     -------
   203303                     55 files
```

Unlike our previous mechanism, we can't actually choose where each file goes in this output tree, except by moving them around in our source tree. The `data` parameter is part of calculating the transitive closure of dependencies that get included; everything we put here will end up in the final runfiles tree. So we use it to pull in all of the things like our `kubernetes.yaml` files and our `GIT_VERSION`/`BRANCH_NAME`/`DOWNLOAD_URL` files and our `spire/resources/*` files et cetera.